From 4cec6b20896760e115207ded225ca7a3b530f842 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Fri, 22 Feb 2013 14:23:02 +0800
Subject: [PATCH 163/281] ENGR00242670 fsl: fix sdcard storage not work issue
 on HDMI dongle

fix sdcard storage not work issue on HDMI dongle

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 hdmidongle_6dq/BoardConfig.mk  | 9 +++++++++
 hdmidongle_6dq/init.rc         | 5 +----
 hdmidongle_6dq/recovery.fstab  | 2 +-
 hdmidongle_6dq/ubi/ubinize.ini | 8 ++++++++
 hdmidongle_6dq/vold.fstab      | 2 +-
 5 files changed, 20 insertions(+), 6 deletions(-)

diff --git a/hdmidongle_6dq/BoardConfig.mk b/hdmidongle_6dq/BoardConfig.mk
index f185f80..f0bc64c 100755
--- a/hdmidongle_6dq/BoardConfig.mk
+++ b/hdmidongle_6dq/BoardConfig.mk
@@ -64,6 +64,15 @@ TARGET_UBIRAW_ARGS := -m 4096 -p 1024KiB $(UBI_ROOT_INI)
 # for Micron MT29F32G08MAA
 #TARGET_MKUBIFS_ARGS := -m 4096 -e 516096 -c 4096 -x none
 #TARGET_UBIRAW_ARGS := -m 4096 -p 512KiB $(UBI_ROOT_INI)
+
+# Note: this NAND partition table must align with MFGTool's config.
+BOARD_KERNEL_CMDLINE +=  mtdparts=gpmi-nand:16m(bootloader),16m(bootimg),128m(recovery),-(root) ubi.mtd=3
+endif
+
+ifeq ($(TARGET_USERIMAGES_USE_UBIFS),true)
+ifeq ($(TARGET_USERIMAGES_USE_EXT4),true)
+$(error "TARGET_USERIMAGES_USE_UBIFS and TARGET_USERIMAGES_USE_EXT4 config open in same time, please only choose one target file system image")
+endif
 endif
 
 # atheros 3k BT
diff --git a/hdmidongle_6dq/init.rc b/hdmidongle_6dq/init.rc
index 291a491..56904c7 100755
--- a/hdmidongle_6dq/init.rc
+++ b/hdmidongle_6dq/init.rc
@@ -111,10 +111,7 @@ service iprenew_wlan0 /system/bin/dhcpcd -n
 
 on fs
 # mount ext4 partitions
-    mount ubifs ubi0:system /system
-    mount ubifs ubi0:system /system ro remount
-    mount ubifs ubi0:data /data
-    mount ubifs ubi0:cache /cache
+    mount_all /fstab.freescale
 
 # for compat with compat-wireless
     insmod  /system/lib/modules/extra/cfg80211.ko
diff --git a/hdmidongle_6dq/recovery.fstab b/hdmidongle_6dq/recovery.fstab
index 1849f7f..0e9e3ca 100644
--- a/hdmidongle_6dq/recovery.fstab
+++ b/hdmidongle_6dq/recovery.fstab
@@ -1,5 +1,5 @@
 # Use below lines if you use NAND:
-/boot mtd bootimg
+/boot mtd /dev/mtd/mtd1
 /recovery mtd /dev/mtd/mtd2
 /system ubifs /dev/ubi0_0
 /data ubifs /dev/ubi0_2
diff --git a/hdmidongle_6dq/ubi/ubinize.ini b/hdmidongle_6dq/ubi/ubinize.ini
index 7df602d..3cb6ff5 100644
--- a/hdmidongle_6dq/ubi/ubinize.ini
+++ b/hdmidongle_6dq/ubi/ubinize.ini
@@ -23,3 +23,11 @@ vol_type=dynamic
 vol_name=data
 vol_alignment=1
 vol_flags=autoresize
+
+[device]
+mode=ubi
+vol_id=3
+vol_size=20MiB
+vol_type=dynamic
+vol_name=device
+vol_aligment=1
diff --git a/hdmidongle_6dq/vold.fstab b/hdmidongle_6dq/vold.fstab
index db88190..dffc61e 100644
--- a/hdmidongle_6dq/vold.fstab
+++ b/hdmidongle_6dq/vold.fstab
@@ -28,7 +28,7 @@ dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
 #mount SDHC4 SD card /mnt/sdcard as primary storage forMX6Q SABER_LITE RevC
 dev_mount extsd /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.1/mmc_host/mmc1
 #mount SDHC3 TF card to /mnt/extsd as external storage forMX6Q SABER_LITE RevC
-dev_mount sdcard /mnt/sdcard 1 /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc0
+#dev_mount sdcard /mnt/sdcard 1 /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc0
 
 ## Example of a dual card setup
 # dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-- 
1.8.0

